src/SchedulerSubsystem/Scheduler.java

```diff
@@ -0,0 +1,45 @@
+package SchedulerSubsystem;
+import ElevatorSubsytem.Elevator;
+
+import java.util.ArrayList;
+
+public class Scheduler implements Runnable {
+    private ArrayList<SchedulerEvent> eventQueue;
+    private ArrayList<Thread> elevators;
+    private ArrayList<Thread> floors;
+
+    public Scheduler() {
+    }
+
+    public void setFloors(ArrayList<Thread> floors) {
+        this.floors = floors;
+    }
+
+    public void setElevators(ArrayList<Thread> elevators) {
+        this.elevators = elevators;
+    }
+
+    /**
+     * Receives passenger event from floor and creates scheduler event
+     * @param schedulerEvent
+     * @return
+     */
+    public SchedulerEvent receivePassengerRequest(SchedulerEvent schedulerEvent) {
```

Comment:
This should store it internally in a queue instead of returning an event. 

Also, is it possible to implement this through inheriting a parent "Receiver" Interface? We can subclass the interface to match our event types as well. We'll probably re-use it for floor and Elevator. All the receiver would need is receive(EventType event) -> event type would depend on the receiver subclass (i.e. DestinationReceiver).